server/src/main/java/keywhiz/service/daos/SecretSeriesDAO.java

```diff
@@ -201,18 +217,21 @@ public int setHmac(long secretContentId, String hmac) {
   }
 
   public int setCurrentVersion(long secretId, long secretContentId, String updater, long now) {
-    long checkId;
-    Record1<Long> r = dslContext.select(SECRETS_CONTENT.SECRETID)
+    SecretsContentRecord r = dslContext
+        .select(
+            SECRETS_CONTENT.SECRETID,
+            SECRETS_CONTENT.EXPIRY)
         .from(SECRETS_CONTENT)
         .where(SECRETS_CONTENT.ID.eq(secretContentId))
-        .fetchOne();
+        .fetchOneInto(SECRETS_CONTENT);
```

Comment:
Although it is possible to select arbitrary column values into a generic record, I think it's a lot more clear to always map the results into a properly typed record, even if not every field is populated.